web/impact/impact/v1/events/organization_became_finalist_event.py

```diff
@@ -21,23 +35,23 @@ def events(cls, organization):
                 result.append(cls(ss))
         return result
 
-    def serialize(self):
-        finalist_time = self._finalist_datetime()
-        program = self.startup_status.program_startup_status.program
-        return {
-            "cycle": program.cycle.name,
-            "cycle_id": program.cycle.id,
-            "datetime": finalist_time,
-            "description": self.DESCRIPTION_FORMAT.format(
-                self.startup_status.program_startup_status.program.name),
-            "event_type": self.EVENT_TYPE,
-            "program": program.name,
-            "program_id": program.id,
-            }
-
-    def _finalist_datetime(self):
-        result = self.startup_status.created_at
-        if result is None:
-            program = self.startup_status.program_startup_status.program
-            return program.start_date
-        return result
+    def calc_datetimes(self):
+        self.earliest = self.startup_status.created_at
+        if self.earliest is None:
+            self.earliest = self._program.start_date
+
+    def description(self):
+        return self.DESCRIPTION_FORMAT.format(
+            self.startup_status.program_startup_status.program.name)
+
+    def cycle(self):
+        return self._program.cycle.name
+
+    def cycle_id(self):
```

Comment:
it seems like a lot of these do the exact same thing (i see ```def cycle_id``` duplicated in several places. Can we pull these into a base class?

Reply:
Pushing back on this.  The implementation of cycle and cycle_id differ in the two organization classes and that's intentional (entrants are for cycles and finalists are associated with programs).  There is very little additional duplication.